src/main.cpp

```diff
@@ -2809,6 +2809,7 @@ bool AcceptBlockHeader(const CBlockHeader& block, CValidationState& state, CBloc
     uint256 hash = block.GetHash();
     BlockMap::iterator miSelf = mapBlockIndex.find(hash);
     CBlockIndex *pindex = NULL;
+    if (hash != chainparams.GetConsensus().hashGenesisBlock) {
     if (miSelf != mapBlockIndex.end()) {
```

Comment:
Yes, that is still the plan at some point, although I'd say it's very low priority.

Reducing the diff is generally a good reason, but not enough to deviate this much from the coding style IMO. This makes the control flow harder to interpret in an important function.